generator/src/main/java/com/scottlogic/deg/generator/fieldspecs/FieldSpecMerger.java

```diff
@@ -56,6 +57,28 @@
         }
 
         //operation/s that must happen last
-        return finalMergeOperation.applyMergeOperation(left, right, merging.get());
+        Optional<FieldSpec> fieldSpec = finalMergeOperation.applyMergeOperation(left, right, merging.get());
+
+        if (!fieldSpec.isPresent() || isContradictory(fieldSpec.get())){
+            return Optional.empty();
+        }
+
+        return fieldSpec;
+    }
+
+    private Boolean isContradictory(FieldSpec fieldSpec){
```

Comment:
a few points here:
1. Minor point, should this return `boolean` (the non-nullable type)?
1. Does this implementation solve #475
1. Does the name `isContradictory` clarify what the method does, could it be better named, e.g. `cannotEmitAnyData`